lib/jekyll-gist/gist_tag.rb

```diff
@@ -64,20 +71,43 @@ def gist_noscript_tag(gist_id, filename = nil)
       end
 
       def fetch_raw_code(gist_id, filename = nil)
+        return code_from_api(gist_id, filename) if ENV["JEKYLL_GITHUB_TOKEN"]
+
         url = "https://gist.githubusercontent.com/#{gist_id}/raw"
         url = "#{url}/#{filename}" unless filename.empty?
-        begin
-          uri = URI(url)
-          Net::HTTP.start(uri.host, uri.port,
-            use_ssl: uri.scheme == 'https',
-            read_timeout: 3, open_timeout: 3) do |http|
-            request = Net::HTTP::Get.new uri.to_s
-            response = http.request(request)
-            response.body
-          end
-        rescue SocketError, Net::HTTPError, Net::OpenTimeout, Net::ReadTimeout, TimeoutError
-          nil
+        uri = URI(url)
+        Net::HTTP.start(uri.host, uri.port,
+          use_ssl: uri.scheme == 'https',
+          read_timeout: 3, open_timeout: 3) do |http|
+          request = Net::HTTP::Get.new uri.to_s
+          response = http.request(request)
+          response.body
+        end
+      rescue SocketError, Net::HTTPError, Net::OpenTimeout, Net::ReadTimeout, TimeoutError
+        nil
+      end
+
+      private
+
+      def code_from_api(gist_id, filename = nil)
+        gist = client.gist gist_id
+
+        file = if filename.to_s.empty?
+          # No file specified, return the value of the first key/value pair
+          gist.files.first[1]
+        else
+          # .files is a hash of :"filename.extension" => data pairs
+          # Rather than using to_sym on arbitrary user input,
+          # Find our file by calling to_s on the keys
+          match = gist.files.find { |name, data| name.to_s == filename }
+          match[1] if match
         end
+
+        file[:content] if file
+      end
+
+      def client
+        @client ||= Octokit::Client.new :access_token => ENV["JEKYLL_GITHUB_TOKEN"]
```

Comment:
Whoops, I'm awfully silly. Really sorry, Ben, I misspoke. When I asked about caching, my reasoning was that for each `{% gist %}`, a new `GistTag` instance is created and thus requires its own client. The Client won't change between various `gist` calls, so storing it so it's on the class rather than instance (or similar, store it anywhere you feel is appropriate) gives us a little speed boost and relieves some pressure on the garbage collector.

Does that make sense? Sorry I didn't communicate all that. Still figuring out the right balance between being concise (saves time) and communicating 100% of what I was thinking.

So this code would become:

```ruby
def self.client
  @client ||= Octokit::Client.new #...
end
```

and would be accessed via `GistTag.client` (i.e. at the class level).